src/applications/vaos/referral-appointments/ChooseDateAndTime.jsx

```diff
@@ -59,12 +68,63 @@ export const ChooseDateAndTime = () => {
   };
   const tzLong = getTzName('longGeneric');
   const tzShort = getTzName('shortGeneric');
+  useEffect(
+    () => {
+      if (selectedDate && upcomingAppointments) {
```

Comment:
It might be nice to create a utility method for this called `hasConflict` or something that returns a boolean that way we can unit test that method and it could be a little cleaner and easier to read.

We could also within that method do stuff like:
```
if (!selectedDate || !upcomingAppointments) return false;
if (!daysWithApts.includes(selectedDay)) return false;
```
so less nesting.

Reply:
makes sense to me, see refactor.